src/test/scala/org/broadinstitute/dsde/workbench/sam/model/ResourceTypeSpec.scala

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.dsde.workbench.sam.model
+
+import org.broadinstitute.dsde.workbench.sam.Generator
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+
+class ResourceTypeSpec extends AnyFlatSpecLike with Matchers {
+  behavior of "ResourceType.getRoleActions"
+
+  it should "return the correct actions for a resource type" in {
+    val resourceType = Generator.genResourceType.sample.get
+    resourceType.roles.foreach { role =>
+      resourceType.getRoleActions(role.roleName) should be(role.actions)
+    }
+  }
+
+  it should "return the correct actions for a resource type with includedRoles" in {
+    val resourceType = Generator.genResourceType.sample.get
+    val testRole = ResourceRoleName("includedRole")
+    val withIncludedRoles = resourceType.copy(
+      roles = resourceType.roles + ResourceRole(testRole, Set.empty, resourceType.roles.map(_.roleName))
+    )
```

Comment:
if I'm reading this correctly, the includedRoles' actions are an exact duplicate of the non-included roles' actions, so this test doesn't really exercise the recursion. Could you use a different role/actions for the included ones?

Reply:
This should exercise the recursion. `testRole` is not given any direct actions, it only includes other roles. So any actions it has come from the recursion.